evm-contracts/src/v0.7/ChainlinkClient.sol

```diff
@@ -0,0 +1,302 @@
+    bytes32 _specId,
+    address _callbackAddress,
+    bytes4 _callbackFunctionSignature
+  ) internal pure returns (Chainlink.Request memory) {
+    Chainlink.Request memory req;
+    return req.initialize(_specId, _callbackAddress, _callbackFunctionSignature);
+  }
+
+  /**
+   * @notice Creates a Chainlink request to the stored oracle address
+   * @dev Calls `chainlinkRequestTo` with the stored oracle address
+   * @param _req The initialized Chainlink Request
+   * @param _payment The amount of LINK to send for the request
+   * @return requestId The request ID
+   */
+  function sendChainlinkRequest(Chainlink.Request memory _req, uint256 _payment)
+    internal
+    returns (bytes32)
+  {
+    return sendChainlinkRequestTo(address(oracle), _req, _payment);
+  }
+
+  /**
+   * @notice Creates a Chainlink request to the stored oracle address
+   * @dev Calls `chainlinkRequestTo` with the stored oracle address
+   * @param _req The initialized Chainlink Request
+   * @param _payment The amount of LINK to send for the request
+   * @return requestId The request ID
+   */
+  function sendChainlinkRequest2(Chainlink.Request memory _req, uint256 _payment)
+    internal
+    returns (bytes32)
+  {
+    return sendChainlinkRequest2To(address(oracle), _req, _payment);
+  }
+
+  /**
+   * @notice Creates a Chainlink request to the specified oracle address
+   * @dev Generates and stores a request ID, increments the local nonce, and uses `transferAndCall` to
+   * send LINK which creates a request on the target oracle contract.
+   * Emits ChainlinkRequested event.
+   * @param _oracle The address of the oracle for the request
+   * @param _req The initialized Chainlink Request
+   * @param _payment The amount of LINK to send for the request
+   * @return requestId The request ID
+   */
+  function sendChainlinkRequestTo(address _oracle, Chainlink.Request memory _req, uint256 _payment)
+    internal
+    returns (bytes32 requestId)
+  {
+    requestId = keccak256(abi.encodePacked(this, requestCount));
+    _req.nonce = requestCount;
+    pendingRequests[requestId] = _oracle;
+    emit ChainlinkRequested(requestId);
+    require(link.transferAndCall(_oracle, _payment, encodeRequest(_req, ARGS_VERSION_1)), "unable to transferAndCall to oracle");
+    requestCount += 1;
+
+    return requestId;
+  }
+
+  /**
+   * @notice Creates a Chainlink request version 2 to the specified oracle address
+   * @dev Generates and stores a request ID, increments the local nonce, and uses `transferAndCall` to
+   * send LINK which creates a request on the target oracle contract.
+   * Emits ChainlinkRequested event.
+   * @param _oracle The address of the oracle for the request
+   * @param _req The initialized Chainlink Request
+   * @param _payment The amount of LINK to send for the request
+   * @return requestId The request ID
+   */
+  function sendChainlinkRequest2To(address _oracle, Chainlink.Request memory _req, uint256 _payment)
```

Comment:
I don't love this name @se3000 ... Does there need to be 2 separate functions here? Because it means that we need:

- `sendChainlinkRequest`
- `sendChainlinkRequest2`
- `sendChainlinkRequestTo`
- `sendChainlinkRequest2To`

Could we just keep `sendChainlinkRequest` and `sendChainlinkRequestTo`, and add a parameter for `dataVersion`.

That would require knowledge by the consumer, however.

Reply:
Removed these, forcing `dataVersion` 2 in solidity v0.7 Operator.